app/server.go

```diff
@@ -1096,6 +1072,15 @@ func (s *Server) Shutdown() {
 
 	mlog.Info("Server stopped")
 
+	// Stop products.
+	// This needs to happen at last because products are dependent
+	// on parent services.
+	for name, product := range s.products {
+		if err2 := product.Stop(); err2 != nil {
```

Comment:
Is it feasible to avoid `err2` in favour of `appErr` or whatever is necessary in context?

Reply:
The `Stop` method does indeed return an `error`. I am 0/5 on any other variable name if you prefer, but we need to keep using `:=` to avoid bugs. That's a real scenario and has happened in the past.